egs/wsj/s5/steps/nnet3/multilingual/get_egs.sh

```diff
@@ -0,0 +1,158 @@
+  echo "Usage: $0 [opts] num-input-langs <data-dir-per-lang> <ali-dir-per-lang> <egs-dir-per-lang> <multilingual-egs-dir>"
+  echo " e.g.: $0 2 data/lang1/train data/lang2/train "
+       " exp/lang1/tri5_ali exp/lang2/tri5_ali exp/lang1/nnet3/lang1 exp/lang2/nnet3/lang2 exp/multi/egs"
+  echo ""
+  echo "Main options (for others, see top of script file)"
+  echo "  --config <config-file>                           # config file containing options"
+  echo "  --num-jobs <nj>                                        # The maximum number of jobs you want to run in"
+  echo "                                                   # parallel (increase this only if you have good disk and"
+  echo "                                                   # network speed).  default=6"
+  echo "  --cmd (utils/run.pl;utils/queue.pl <queue opts>) # how to run jobs."
+  echo "  --samples-per-iter <#samples;400000>             # Target number of egs per archive (option is badly named)"
+  echo "  --frames-per-eg <frames;8>                       # number of frames per eg on disk"
+  echo "  --left-context <width;4>                         # Number of frames on left side to append for feature input"
+  echo "  --right-context <width;4>                        # Number of frames on right side to append for feature input"
+  echo "  --num-frames-diagnostic <#frames;4000>           # Number of frames used in computing (train,valid) diagnostics"
+  echo "  --num-valid-frames-combine <#frames;10000>       # Number of frames used in getting combination weights at the"
+  echo "                                                   # very end."
+  echo "  --stage <stage|0>                                # Used to run a partially-completed training process from somewhere in"
+  echo "                                                   # the middle."
+
+  exit 1;
+fi
+
+num_lang=$1
+shift
+args=("$@")
+megs_dir=${args[-1]}
+num_lang_args=$[$[${#args[@]}-1]/3] 
+
+if [ ${#args[@]} != $[$num_lang*3+1] ]; then
+  echo "$0: num of input dirs provided for all langs is not compatible with num-langs in input." && exit 1;
+fi
+
+# read input data, ali and egs dir per lang
+for l in `seq 0 $[$num_lang-1]`; do
+  multi_data_dirs[$l]=${args[$l]}
+  multi_ali_dirs[$l]=${args[$l+$num_lang]}
+  multi_egs_dirs[$l]=${args[$l+2*$num_lang]}
+done
+
+if [ $stage -le 0 ];then
+  echo "$0: Generate separate egs directory per language for multilingual training."
+  echo num_langs = $num_lang
+  for lang in `seq 0 $[$num_lang-1]`; do
+    data=${multi_data_dirs[$lang]} 
+    ali_dir=${multi_ali_dirs[$lang]}
+    egs_dir=${multi_egs_dirs[$lang]}
+    online_ivector_dir=
+    if [ ! -z "$multi_ivector_dirs" ]; then
+      online_ivector_dir=${multi_ivector_dirs[$lang]}
+    fi
+    #if [ ! -d "$egs_dir" ]; then
+    if true; then
+      echo "$0: Generate egs for ${lang_list[$lang]}"
+      if [[ $(hostname -f) == *.clsp.jhu.edu ]] && [ ! -d $egs_dir/storage ]; then
+        utils/create_split_dir.pl \
```

Comment:
this create_split_dir stuff is not the kind of thing that should appear in scripts inside steps/.